IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -387,42 +442,111 @@ - (BOOL)removeAccountsWithKey:(MSIDCacheKey *)key
 #pragma mark - Credentials
 
 // Write a credential to the macOS keychain cache.
-- (BOOL)saveToken:(__unused MSIDCredentialCacheItem *)credential
-              key:(__unused MSIDCacheKey *)key
-       serializer:(__unused id<MSIDCacheItemSerializing>)serializer
-          context:(__unused id<MSIDRequestContext>)context
-            error:(__unused NSError **)error
+- (BOOL)saveToken:(MSIDCredentialCacheItem *)credential
+              key:(MSIDCacheKey *)key
+       serializer:(id<MSIDCacheItemSerializing>)serializer
+          context:(id<MSIDRequestContext>)context
+            error:(NSError **)error
 {
     assert(credential);
     assert(serializer);
+    
     [self updateLastModifiedForCredential:credential context:context];
-    NSData *itemData = [serializer serializeCredentialCacheItem:credential];
+    NSString *credentialKey = [NSString stringWithFormat:@"%@%@%@", key.account, keyDelimiter, key.service];
```

Comment:
I also saw that shared credential puts this key string together. Can we have one common place for it, e.g. in the key itself perhaps? or pass the key obj itself to shared credential, and let it deal with it.

Reply:
Updated code to pass MSIDDefaultCredentialKey to credentialStorage item.